xbmc/settings/AdvancedSettings.cpp

```diff
@@ -1406,3 +1407,19 @@ void CAdvancedSettings::setExtraLogLevel(const std::vector<CVariant> &components
     m_extraLogLevels |= static_cast<int>(it->asInteger());
   }
 }
+
+std::string CAdvancedSettings::GetMusicExtensions() const
```

Comment:
this function is basically called once every time we list a directory. i estimate it's gonna have a runtime in the order of nanosecs and drowns completely in IO which is on the order of millisecs at the very least (more likely in the order of seconds). that's a 3 order difference in runtime minimum. a cache means logic to maintain that cache if an addon is added, removed etc. 

not worth it imo and nothing i will invest time in. you are free to do so of course.